tidy3d/plugins/microwave/path_integrals.py

```diff
@@ -0,0 +1,299 @@
+            scalar_field = scalar_field.drop_sel({coord: (min_bound, max_bound)}, errors="ignore")
+            coordinates = scalar_field.coords[coord].values
+        else:
+            coordinates = scalar_field.coords[coord].sel({coord: slice(min_bound, max_bound)})
+
+        # Integration is along the original coordinates plus ensure that
+        # endpoints corresponding to the precise bounds of the port are included
+        coords_interp = np.array([min_bound])
+        coords_interp = np.concatenate((coords_interp, coordinates))
+        coords_interp = np.concatenate((coords_interp, [max_bound]))
+        coords_interp = {coord: coords_interp}
+
+        # Use extrapolation for the 2 additional endpoints, unless there is only a single sample point
+        method = "linear"
+        if len(coordinates) == 1 and self.extrapolate_to_endpoints:
+            method = "nearest"
+        scalar_field = scalar_field.interp(
+            coords_interp, method=method, kwargs={"fill_value": "extrapolate"}
+        )
+        return scalar_field.integrate(coord=coord)
+
+    def _get_field_along_path(
+        self, scalar_field: AbstractSpatialDataArray
+    ) -> AbstractSpatialDataArray:
+        """Returns a selection of the input `scalar_field` ready for integration."""
+        axis1 = self.remaining_axes[0]
+        axis2 = self.remaining_axes[1]
+        coord1 = "xyz"[axis1]
+        coord2 = "xyz"[axis2]
+
+        if self.snap_path_to_grid:
+            # Coordinates that are not integrated
+            remaining_coords = {
+                coord1: self.center[axis1],
+                coord2: self.center[axis2],
+            }
+            # Select field nearest to center of integration line
+            scalar_field = scalar_field.sel(
+                remaining_coords,
+                method="nearest",
+                drop=False,
+            )
+        else:
+            # Try to interpolate unless there is only a single coordinate
+            coord1dict = {coord1: self.center[axis1]}
+            if scalar_field.sizes[coord1] == 1:
+                scalar_field = scalar_field.sel(coord1dict, method="nearest")
+            else:
+                scalar_field = scalar_field.interp(
+                    coord1dict, method="linear", kwargs={"bounds_error": True}
+                )
+            coord2dict = {coord2: self.center[axis2]}
+            if scalar_field.sizes[coord2] == 1:
+                scalar_field = scalar_field.sel(coord2dict, method="nearest")
+            else:
+                scalar_field = scalar_field.interp(
+                    coord2dict, method="linear", kwargs={"bounds_error": True}
+                )
+
+        return scalar_field
+
+    @cached_property
+    def main_axis(self) -> Axis:
+        """Axis for performing integration."""
+        val = next((index for index, value in enumerate(self.size) if value != 0), None)
+        return val
+
+
+class VoltageIntegralAA(AxisAlignedPathIntegral):
```

Comment:
I'm not 100% sold on `AA` in the class name. Do you think it would be very clear to the user that this means `AxisAligned`? Would `VoltageIntegralAxisAligned` be too long of a class name?

Reply:
I agree, I have switched to the full expansion.